src/App.tsx

```diff
@@ -15,35 +16,98 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  BY_LENGTH = 'length',
+  ALPHABETICALLY = 'alphabetically',
+  DEFAULT = '',
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState<SortType>(SortType.DEFAULT);
+  const [sortOrder, setSortOrder] = useState<boolean>(true);
+
+  function prepareGoods(goods: string[], sortBy: string, order = true) {
+    const preparedGoods = [...goods];
+
+    switch (sortBy) {
+      case SortType.BY_LENGTH: {
+        preparedGoods.sort((goodA, goodB) => goodA.length - goodB.length);
+        break;
+      }
+
+      case SortType.ALPHABETICALLY: {
+        preparedGoods.sort((goodA, goodB) => goodA.localeCompare(goodB));
+        break;
+      }
+
+      default:
+        break;
+    }
+
+    return order ? preparedGoods : preparedGoods.reverse();
+  }
+
+  function resetHandler(): void {
+    setSortField(SortType.DEFAULT);
+    setSortOrder(true);
```

Comment:
The function `resetHandler` should be renamed to start with a verb to follow the [CODE STYLE] convention. A better name could be `handleReset` to express the action it performs.